doc/design/speech/README.MD

```diff
@@ -140,7 +140,17 @@ TODO by Assignees
 
 ### Beam Search with CTC and LM
 
-TODO by Assignees
+<div align="center">
+<img src="image/beam_search.png" width=400><br/>
+Figure 2. Algorithm for Beam Search Decoder.
+</div>
+
+- The **Beam Search Decoder** for DS2 CTC-trained network follows the similar approach in \[[3](#references)\] with a modification for the ambiguous part, as shown in Figure 2.
```

Comment:
What is the ambiguous part and what is the modification? Could you please add more details?

Reply:
Done